core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -414,7 +414,7 @@ private[spark] class AppStatusListener(
         val e = it.next()
         if (job.stageIds.contains(e.getKey()._1)) {
           val stage = e.getValue()
-          if (v1.StageStatus.PENDING.equals(stage.status)) {
+          if (v1.StageStatus.PENDING.equals(stage.status) && stage.info.numTasks > 0) {
```

Comment:
Shall we change `stageId == -1`? Of course, CONSTANT will be better like `stageId == DUMMY_ID`.

Reply:
Changes in this PR is to care the case when a job has 1 or more partitions (tasks).
So, I think that `numTasks > 0` makes sense rather than `stageId != -1` .
Or how about making the `while` loop  containing this code block like `while (job.stageIds.nonEmpty && it.hasNext())`
What do you think?

Reply:
FYI:
This change is to avoid marking a stage which has no partitions as skipped.
<img width="1436" alt="スクリーンショット 2019-11-26 17 12 15" src="https://user-images.githubusercontent.com/4736016/69615082-a4304180-1077-11ea-9459-791d8074852b.png">